spec/stac-geoparquet-spec.md

```diff
@@ -0,0 +1,88 @@
+[GeoParquet](https://geoparquet.org). It is directly inspired by the [STAC GeoParquet](https://github.com/stac-utils/stac-geoparquet)
+library, but aims to provide guidance for anyone putting STAC data into GeoParquet. 
+
+## Use cases
+
+* Provide a STAC GeoParquet that mirrors a static Collection as a way to query the whole dataset instead of reading every specific GeoJSON file.
+* As an output format for STAC API responses that is more efficient than paging through thousands of pages of GeoJSON.
+* Provide efficient access to specific fields of a STAC item, thanks to Parquet's columnar format.
+
+## Guidelines
+
+Each row in the Parquet Dataset represents a single STAC item. Most all the fields in a STAC Item should be mapped to a column in GeoParquet. We embrace Parquet structures where possible, mapping
+from JSON into nested structures. We do pull the properties to the top level, so that it is easier to query and use them. The names of
+most of the fields should be the same in STAC and in GeoParquet.
+
+| Field              | GeoParquet Type      | Required | Details                                                                                                                        |
+|--------------------|----------------------|----------|--------------------------------------------------------------------------------------------------------------------------------|
+| type               | String               | Optional | This is just needed for GeoJSON, so it is optional and not recommended to include in GeoParquet                                |
+| stac_extensions    | List of Strings      | Required | This column is required, but can be empty if no STAC extensions were used                                                      |
+| id                 | String               | Required | Required, should be unique within each collection                                                                              |
+| geometry           | Binary (WKB)         | Required | For GeoParquet 1.0 this must be well-known Binary                                                                              |
+| bbox               | Struct of Floats     | Required | Can be a 4 or 6 value struct, depending on dimension of the data                                                               |
+| links              | List of Link structs | Required | See [Link Struct](#link-struct) for more info                                                                                  |
+| assets             | An Assets struct     | Required | See [Asset Struct](#asset-struct) for more info                                                                                |
+| collection         | String               | Required | The ID of the collection this Item is a part of                                                                                |
```

Comment:
Does this field make sense to be required? In the usual case where you have all items from one collection, it's pretty much the same as with type. A column with all the same values.

Reply:
Good point. I don't know how to treat this one.

It can technically vary if we're just *strongly recommending* a single collection per datasets, which differentiates it a bit from `type`.

Reply:
For both this and the `type`, I'd recommend we remove them from the table. My thoughts:

1. Users can add additional fields if they want (and will, for columns coming from STAC extensions; we should document this).
2. Consumers of stac-geoparquet who wish to generate STAC items will already need custom code (to move columns to properties). Adding `type` and extracting a collection ID out of the parquet metadata doesn't seem like too much extra work.

By removing `collection` as a *required* field and putting collection metadata in the geoparquet metadata, we really are focusing on a single parquet dataset per collection. I think people are aligned on that, but wanted to confirm.

Reply:
I think collection is fine to be optional, in case there are situations where it's hard for people to write parquet file metadata (e.g. from data warehouses). But in most situations we can say: as long as you have the `Collection` in the file metadata, and you only have one collection, you don't need it as a column.

Reply:
> Users can add additional fields if they want (and will, for columns coming from STAC extensions; we should document this).

Documentation on how to extend this spec would be much appreciated! I've been looking into using stac-geoparquet to convert[ label extension](https://github.com/stac-extensions/label) items to ML ready label datasets, using [EuroSat-STAC](https://github.com/phelber/EuroSAT) as an example.

Reply:
I think we can update this document with additional sections for each extension that gives the name of the field and the parquet type.